### PR TITLE
Fix VLOOKUP whose return was impacted by order for a $not_exact_match

### PR DESCRIPTION
**Pull request by frantzmiccoli**
This is:

```
- [x] a bugfix
- [ ] a new feature
```

Checklist:

- [x] Changes are covered by unit tests
- [x] Code style is respected
- [x] Commit message explains **why** the change is made (see https://github.com/erlang/otp/wiki/Writing-good-commit-messages)
- [x] CHANGELOG.md contains a short summary of the change
- [x] Documentation is updated as necessary

### Why this change is needed?

A bug was noticed where the VLOOKUP was impacted by the ordering of inputs. I noticed this after a call to LOOKUP.

### CODE REVIEW

**Comment by frantzmiccoli** [comment]
I am still investigating the behavior in Excel and trying to be sure that I am fixing this the right way. I will let you know what I am done.

**Comment by frantzmiccoli** [comment]
I have observed inconsistencies between what is implemented and two line examples (mentioned in the unit tests). I don't understand why Microsoft Excel for Mac 15.35 (170610) behaves how it does. LibreOffice 6.0.3.2 however is consistent with this implementation.

I attached a file 
[lookup-reproduction.xlsx](https://github.com/PHPOffice/PhpSpreadsheet/files/2773812/lookup-reproduction.xlsx) if you wish to give a look to this.

File: src/PhpSpreadsheet/Calculation/LookupRef.php [hunk 1]
```diff
@@ -719,7 +719,6 @@ public static function VLOOKUP($lookup_value, $lookup_array, $index_number, $not
                 (!is_numeric($lookup_value) && !is_numeric($rowData[$firstColumn]))) {
                 if ($not_exact_match) {
                     $rowNumber = $rowKey;
-                    $rowValue = $rowData[$firstColumn];
 
                     continue;
                 } elseif ((strtolower($rowData[$firstColumn]) == strtolower($lookup_value))
```
Comment (MarkBaker) [inline src/PhpSpreadsheet/Calculation/LookupRef.php]: Not related to the bug that you're fixing, but it's probably worth changing these from `strtolower()` to use the `StringHelper::strToLower()` to make them multibyte-safe

Comment (frantzmiccoli) [reply src/PhpSpreadsheet/Calculation/LookupRef.php]: Will try to do that over the week-end.

**Comment by frantzmiccoli** [comment]
@MarkBaker done, this fixes the inconsistencies with Excel.

**Pull request merged**